internal/utils.go

```diff
@@ -3,9 +3,37 @@ package internal
 import (
 	"fmt"
 	"net/url"
+	"runtime"
+	"runtime/debug"
 	"strings"
 )
 
+var (
+	// SdkVersion could be set via build flags `-ldflags="-X internal.SdkVersion=${CI_COMMIT_TAG:-$CI_COMMIT_SHA}"`
```

Comment:
This isn't good Go practice, as no one's going to want to override it this way. If there really needs to be open support for changing the user-agent, then it should be via a publicly available function like the currently-internal `WithSdkIdentifier` that helps to build up the request object with custom parameters, one of which could be the user-agent.